### ISSUE TITLE
Get functional testing to start sending emails 

### ISSUE THREAD

**Issue by MichaelClerx** [pints-team/pints#786]
- On failures at least. Might also be good to notify of changes in distribution later

**Comment by fcooper8472** [comment]
Todo:
- New mailing list `pints-devs` or similar for a subset of people who want to be notified about automated pints tests
- Can GitHub report things via email?
- Is it worth setting up a self-hosted CI tool, like Azure Pipelines, Buildbot or Jenkins?

**Comment by fcooper8472** [comment]
New mailing list is `PINTS Core Devs Mailing List <pints-core-devs@maillist.ox.ac.uk>`

**Comment by MichaelClerx** [comment]
Awesome! Thanks Fergus!

**Comment by fcooper8472** [comment]
After a number of discussions, it seemed like a good plan to move towards a CI system for orchestrating the functional testing.  I also wanted to mess around with azure devops, the new all-in-one system from Microsoft.

What I have done is set up a [pints portal](https://dev.azure.com/FergusCooper/PINTS/) which I can give everyone access to if/when we're happy to pursue this avenue further.

I have set up a worker on skip that communicates directly with azure devops, and you can manually launch builds from the web interface (if you so wish).  The builds are controlled by a file `azure-pipelines.yaml` in the functional testing repo.  This is essentially a translation of the bash script previously run by the cron job, with some modification to the functional testing to report test failures (through printing to stderr).

At the moment a drawback is that you cannot specify scheduled builds in the yaml schema, so for the time being a new cron job is prodding the existing repo which triggers each build each hour, but this will be simplified when the functionality becomes available.

tl;dr:

- We now have a sophisticated CI pipeline for functional testing
- The testing is still running directly on skip, but you can monitor the status of each build from the web address above
- Failures should be automatically reported to the `<pints-core-devs@maillist.ox.ac.uk>` mailing list

**Comment by fcooper8472** [comment]
@martinjrobins @ben18785 might also be interested in these changes.  They might not be coherent, so please ask any questions!

**Comment by MichaelClerx** [comment]
Awesome! Looking forward to seeing this on Tuesday :-)

**Comment by MichaelClerx** [comment]
Shall we close this now @fcooper8472 ?

**Comment by fcooper8472** [comment]
Yep!
